android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothStandardWeightProfile.java

```diff
@@ -823,29 +805,59 @@ protected void chooseExistingScaleUser(Vector<ScaleUser> userList) {
         chooseScaleUserUi(choices);
     }
 
-    protected String getInitials(String fullName) {
-        if (fullName == null || fullName.isEmpty() || fullName.chars().allMatch(Character::isWhitespace)) {
-            return getDefaultInitials();
-        }
-        return buildInitialsStringFrom(fullName).toUpperCase();
+    protected String getInitials(final String fullName, final int size) {
+        if (0 == size)
+            return StringUtils.EMPTY_STRING;
+
+        if (StringUtils.isNullOrWhitespace(fullName))
+            return getDefaultInitials(size);
+
+        return buildInitialsStringFrom(fullName, size).toUpperCase();
     }
 
-    private String getDefaultInitials() {
+    private String getDefaultInitials(final int size) {
         int userId = this.selectedUser.getId();
-        int userIndex = getUserScaleIndex(userId);
-        return "P" + userIndex + " ";
+        String userIndex = String.valueOf(getUserScaleIndex(userId));
+
+        if (userIndex.equals(StringUtils.EMPTY_STRING))
+            return StringUtils.generateStringWithRepeatingChar(size);
+
+        if (1 == size)
+            return String.valueOf(userIndex.charAt(0));
+
+        StringBuilder stringBuilder = new StringBuilder();
+
+        stringBuilder.append('P');
+
+        for (int i = 0; i < size - 1; i++) {
+            char charToAdd = ' ';
+
+            if (userIndex.length() > i)
+                charToAdd = userIndex.charAt(i);
+
+            stringBuilder.append(charToAdd);
+        }
+
+        return stringBuilder.toString();
     }
 
-    private String buildInitialsStringFrom(String fullName) {
-        String[] name = fullName.trim().split(" +");
-        String initials = "";
-        for (int i = 0; i < 3; i++) {
-            if (i < name.length && name[i] != "") {
-                initials += name[i].charAt(0);
-            } else {
-                initials += " ";
-            }
+    private String buildInitialsStringFrom(final String fullName, final int size) {
+        String[] names = StringUtils.splitByWhitespace(fullName);
+
+        if (null == names || 0 == names.length)
+            return StringUtils.generateStringWithRepeatingChar(size);
+
+        StringBuilder initialsBuilder = new StringBuilder();
+
+        for (int i = 0; i < size; i++) {
+            char charToAdd = ' ';
+
+            if (names.length > i && false == names[i].isEmpty())
```

Comment:
Can you assign anything to `names[i].isEmpty()` ?

Reply:
Yes, both are rvalues. However the `false` statement is simpler to spot as such. I rarely think of which is rvalue because it for into my intuition for some of them and take them as such. This my "mean and lean programming" :)

Reply:
🥇  Line 853, if used org.apache.commons.lang3.StringUtils, the library also contains a `public static final String SPACE`. To see if it can be used

Reply:
marked as unresolved to still have in sight the last comment I have added